### PR TITLE
Allow switching sub-environment inside Manage Environment modal

### LINKED ISSUE #2761: Add ability to see what environment is currently selected, and a way to change environment from Manage Environment modal
**Is your feature request related to a problem? Please describe.**

Currently when managing environments and dealing with sub-environment variables from within the base environment, a user has to close the manage environment modal, to check / switch environments for variables to work.

Here, I have a sub-environment that I have just created (with no way to switch to it, or indicator telling me what is selected) and I am attempting to use the variables defined within my base environment - only to be provided with an error indicated it's not available and no indication as to _why_:

Screen Shot 2020-10-21 at 12 02 00 PM

Below we can see that my sub-environment does have the variables defined:

Screen Shot 2020-10-21 at 12 01 25 PM

To get my base environment to work properly, I must close this modal (breaking my flow) and switch to the sub-environment I just created.

Screen Shot 2020-10-21 at 12 04 29 PM

And now I can continue my flow again:

Screen Shot 2020-10-21 at 12 05 01 PM

**Describe the solution you'd like**

Checkbox / radio selector next to sub-environment to indicate what environment is currently active and let me change it from within the manage environments modal

### PR DESCRIPTION
**Pull request by cobwebsonsale**
Closes #2761

- Added a button next to the sub-environment name to activate it
- Added a filled button next to the current active sub-environment
- Added helpful text in tooltips
- Renamed occurrences of `activeEnvironment` to `selectedEnvironment` in `WorkspaceEnvironmentsEditModal` (to avoid confusion) -- separated in another commit, for easier review/revert.

Caveat:
Could not re-render the currently selected environment in the right-pane after activation, so showing the activated environment instead. The re-render only makes sense when Base Environment is selected, as a different active environment changes the validations on variables. Tried:
- calling `_load`, 
- adding `activeEnvironmentId` to `state` inside `WorkspaceEnvironmentsEditModal` and setting state, and
- deriving `activeEnvironmentId` from `activeWorkspaceMeta` in `Wrapper`, as that is what gets updated in `handleChangeEnvironment`

Demo:
switch-environment

### CODE REVIEW

**Comment by develohpanda** [comment]
Wow, thanks a lot for tackling this @cobwebsonsale! 🤗

**Comment by cobwebsonsale** [comment]
Requesting a review here.

**Comment by develohpanda** [comment]
I'm sorry about the mega delays @cobwebsonsale 😞 I'll try to at least give a prelim review this this week

**Review by develohpanda — APPROVED** [review]
Thank you for the PR, QA'd locally and seems to be working as intended, and thanks for splitting up your work into individual commits which made the rename and feature easier to review. 🙌🏽 

My apologies for the delays. Unfortunately the team is spread quite thin at the moment, evident by the sheer number of open-source PRs we need to get through. We currently have a code-freeze in place for migrating to typescript, but once that is done I'll come back to this PR to merge it.

**Comment by cobwebsonsale** [comment]
> seems to be working as intended

Nice, that's good to know.

And no worries about the delay. I want to contribute to help, not to add stress. xD So please take your time and merge this accordingly. Bumped only so it stays on the radar.

Let me know when you come around to this and there are conflicts, I'm surprised there weren't already.

**Comment by develohpanda** [comment]
Sounds good 👍🏽

This is a note to myself to double-check the caveat you describe above and whether that's a concern right now; with the testing done so far it seemed to be fine

**Comment by dimitropoulos** [comment]
I updated this branch for TypeScript: it seems to be working well - 

https://github.com/dimitropoulos/insomnia/commits/feature/switch-env-ts

The first commit is yours (please let me know if I messed up anything on the revival) and the second commit is to fix some TypeScript errors now that we're actually using types for `react-sortable-hoc`.

Please take a look.  If everything is ok, I can push that branch to this one and update this PR, at which point we should be ready (or nearly ready) to go.

**Comment by develohpanda** [comment]
@dimitropoulos having [`2286466` (#2891)](https://github.com/Kong/insomnia/pull/2891/commits/2286466663fec0b4269162a52d24157aba1580ae) and [`d1c4ae2` (#2891)](https://github.com/Kong/insomnia/pull/2891/commits/d1c4ae2a5d2c039c8446128405bdf4f4e1a3bd3d) separate certainly was helpful in processing this PR, but if that's tricky to retain it's okay. I briefly checked your branch and https://github.com/dimitropoulos/insomnia/commit/c5eff8a791a95a24cb00613a9e246eda9889b46b looks correct.

Please push those changes up to `cobwebsonsale:feature/switch-env` and we'll run through another code-review and QA round.

**Comment by dimitropoulos** [comment]
@develohpanda sure, I un-squashed the rename and pushed to the PR's branch.

File: packages/insomnia-app/app/ui/components/modals/workspace-environments-edit-modal.tsx [hunk 1]
```diff
@@ -79,22 +93,44 @@ const SidebarListItem = SortableElement(
...
+          )}
+        </div>
       </li>
     );
   },
 );
 
-const SidebarList = SortableContainer(
-  ({ environments, activeEnvironment, showEnvironment, changeEnvironmentName }) => (
+interface SidebarListProps extends Omit<SidebarListItemProps, 'environment'> {
```
Comment (develohpanda) [inline packages/insomnia-app/app/ui/components/modals/workspace-environments-edit-modal.tsx:110]: Nice use of `Omit`!

File: packages/insomnia-app/app/ui/components/modals/workspace-environments-edit-modal.tsx [hunk 2]
```diff
@@ -225,7 +261,7 @@ class WorkspaceEnvironmentsEditModal extends PureComponent<Props, State> {
 
     // Unset active environment if it's being deleted
     if (activeEnvironmentId === environment._id) {
-      await handleChangeEnvironment(null);
+      handleChangeEnvironment(null);
```
Comment (develohpanda) [inline packages/insomnia-app/app/ui/components/modals/workspace-environments-edit-modal.tsx:264]: Removal of this await actually shows a bug where `Wrapper._handleChangeEnvironment` function swallows a promise. It should be awaited correctly so we don't proceed to delete an environment until the change is complete.

Can be handled separate to this PR

**Review by develohpanda — APPROVED** [review]
Alright, double checked again and it looks good to me

> Could not re-render the currently selected environment in the right-pane after activation, so showing the activated environment instead. The re-render only makes sense when Base Environment is selected, as a different active environment changes the validations on variables.

I worked through this caveat you described; it's interesting that the re-render doesn't occur but when hovering over a template tag in the base environment the sub environment in that tooltip _does_ change but the resolved value does not. This leads me to believe there must be a bug somewhere and that it should work automatically.

Regardless, I still think what you have is fine for this iteration. :tada:

**Review by reynolek — APPROVED** [review]
Product perspective: Good addition, once merged we should probably do a design pass but that doesnt have to happen right now.

**Pull request merged**